cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/KafkaConnectService.java

```diff
@@ -276,4 +309,52 @@ protected ClusterStatus getKafkaConnectStatus(
       return ClusterStatus.OFFLINE;
     }
   }
+
+  public ApiResponse restartConnector(ClusterConnectorRequest clusterConnectorRequest) {
+    log.info("Into restartConnector clusterConnectorRequest {} ", clusterConnectorRequest);
+
+    String suffixUrl =
+        clusterConnectorRequest.getEnv()
+            + "/connectors"
+            + "/"
+            + clusterConnectorRequest.getConnectorName()
+            + "/"
+            + "restart";
+    if (clusterConnectorRequest.isIncludeFailedTasksOnly()) {
+      suffixUrl = suffixUrl + "?includeTasks=true&onlyFailed=true";
+    } else {
+      suffixUrl = suffixUrl + "?includeTasks=true&onlyFailed=false";
+    }
+
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(suffixUrl, clusterConnectorRequest.getProtocol());
+
+    HttpHeaders headers =
+        clusterApiUtils.createHeaders(
+            clusterConnectorRequest.getClusterIdentification(), KafkaClustersType.KAFKA_CONNECT);
+    headers.set("Content-Type", "application/json");
+    HttpEntity<String> request = new HttpEntity<>(headers);
+    ResponseEntity<String> responseNew;
+    try {
+      responseNew =
+          reqDetails.getRight().postForEntity(reqDetails.getLeft(), request, String.class);
+    } catch (HttpServerErrorException | HttpClientErrorException e) {
+      return buildErrorResponseFromRestException(e, CLUSTER_API_ERR_1);
+    } catch (Exception ex) {
+      return ApiResponse.builder().success(false).message(CLUSTER_API_ERR_1).build();
+    }
+    if (responseNew.getStatusCodeValue() == 201 || responseNew.getStatusCodeValue() == 204) {
+      return ApiResponse.builder().success(true).message(ApiResultStatus.SUCCESS.value).build();
+    } else {
+      return ApiResponse.builder().success(false).message(ApiResultStatus.FAILURE.value).build();
+    }
+  }
+
+  public ApiResponse pauseConnector(ClusterConnectorRequest clusterConnectorRequest) {
+    return null;
+  }
```

Comment:
I think we should probably throw UnsupportedOperationException instead of these two nulls, (Even though they should never be called)